hazelcast/src/main/java/com/hazelcast/internal/networking/nio/NioOutboundPipeline.java

```diff
@@ -229,6 +248,22 @@ public void process() throws Exception {
         }
     }
 
+    private void registerOpWrite() throws IOException {
+        registerOp(OP_WRITE);
+
+        if (writeThroughEnabled && !(Thread.currentThread() instanceof NioThread)) {
```

Comment:
Not a big deal, but it might be better to factor out the `NioThread` check into a separate method just to make sure we are using the same check everywhere. Here `instanceof` is used, in `unschedule` `==` is used.

Reply:
I fixed the code so that instanceof is used instead of ==. The JIT should be smart enough to figure out that it can be optimized